modules/internal/CPtr.chpl

```diff
@@ -377,6 +377,31 @@ module CPtr {
     return sizeof(x);
   }
 
+  /*
+    Return the offset of elements in a record.
+
+    .. warning::
+
+      This method is intended for C interoperability.  To enhance flexibility,
+      it is possible to request the offset of elements within a Chapel record.
+      However, be aware:
+
+      * Chapel types are not necessary stored in contiguous memory
+      * Behavior of ``c_offsetof`` may change
+      * Behavior given a Chapel class type field is not well-defined
+   */
+  proc c_offsetof(type t, param fieldname : string) where isRecordType(t) {
```

Comment:
I suggest declaring the return type explicitly and making it size_t, by analogy to c_sizeof().